### DISCUSSION TITLE
git push: could not resolve host: github.com

### DISCUSSION THREAD

**Discussion by chilianlin87** [community/community#31567] [Other Feature Feedback, Questions, & Ideas]
I use git remote added my url
git config --global user.name "myusername"
git config --global user.email "myemail"
git push -u origin master

always give error
fatal: unable to access 'https://github.com/chilianlin87/my-first-repo.git/': Could not resolve host: github.com

I also tried 
git config --global http.proxy http://proxyuser:proxypwd@proxy.server.com:8080/
does not work for me
I also tried
git config --global --unset http.proxy 
git config --global --unset https.proxy
does not work for me either

Anybody can help me please? I am new to git, complete stuck here!!! Thank you so much.

**Comment by F4NT0** [comment]
hi, did you use the following commands to do the remote?

```shell
git remote add origin url
```
and then in a IDE like intellij or vscode when you use `git push -u origin main` it will ask you what kind of way you want to login.

Github change the way to login to make push to github, se if this help :)

Obs: see if the repo you are trying to send is the master branch or main branch

**Reply by chilianlin87** [reply]
> hi, did you use the following commands to do the remote?
> 
> ```shell
> git remote add origin url
> ```
> 
> and then in a IDE like intellij or vscode when you use `git push -u origin main` it will ask you what kind of way you want to login.
> 
> Github change the way to login to make push to github, se if this help :)
> 
> Obs: see if the repo you are trying to send is the master branch or main branch

Thank you so much for the reply.
Yes I did git remote add origin url
when I git push -u origin master, it does not ask for my password to log in, I don't know why it does not ask 
The repo I am trying to push is the master branch
git branch
* master

**Reply by F4NT0** [reply]
Ok, so if `git push -u origin master` doesn't work, try use `git push –set-upstream origin master`.

Did you copy the url from clone option or the URL from your browser?
![image](https://user-images.githubusercontent.com/18719295/187918431-7c8faaf5-2f79-4485-b76d-02e287695d5c.png)

**Reply by chilianlin87** [reply]
> Ok, so if `git push -u origin master` doesn't work, try use `git push –set-upstream origin master`.
> 
> Did you copy the url from clone option or the URL from your browser? ![image](https://user-images.githubusercontent.com/18719295/187918431-7c8faaf5-2f79-4485-b76d-02e287695d5c.png)

I copied url from clone
here is what did
git remote add origin https://github.com/chilianlin87/my-first-repo.git
I just tried `git push –set-upstream origin master`
still give me the same fatal, did not ask for password either.

**Reply by F4NT0** [reply]
well, or is some antivirus or proxy, but you already tried this. try to install the [Github CLI](https://cli.github.com) and clone the repo on your computer and send the files to the repo. something strange is happening on your computer, sorry if i can't help enough

**Reply by chilianlin87** [reply]
> well, or is some antivirus or proxy, but you already tried this. try to install the [Github CLI](https://cli.github.com) and clone the repo on your computer and send the files to the repo. something strange is happening on your computer, sorry if i can't help enough

Thank you so much for the help. I will update you when it solved. Appreciate.

**Comment by airtower-luna** [comment]
> fatal: unable to access 'https://github.com/chilianlin87/my-first-repo.git/': Could not resolve host: github.com

The message says that Git could not find an IP address for `github.com`. That means there's a fundamental network issue, specifically with DNS.

A few things for debugging:
* Can you reach other internet sites from that machine? Can you reach GitHub in other ways (e.g. ping or SSH)?
* You tried to setup a proxy. If you're on a network where you expect having to go through a proxy, ask the network admins.
* Do you have IPv4 connectivity? Unfortunately GitHub still doesn't support IPv6, see #10539. :confused:

**Reply by chilianlin87** [reply]
> > fatal: unable to access 'https://github.com/chilianlin87/my-first-repo.git/': Could not resolve host: github.com
> 
> The message says that Git could not find an IP address for `github.com`. That means there's a fundamental network issue, specifically with DNS.
> 
> A few things for debugging:
> 
> * Can you reach other internet sites from that machine? Can you reach GitHub in other ways (e.g. ping or SSH)?
> * You tried to setup a proxy. If you're on a network where you expect having to go through a proxy, ask the network admins.
> * Do you have IPv4 connectivity? Unfortunately GitHub still doesn't support IPv6, see [IPv6 support for cloning Git repositories #10539](https://github.com/orgs/community/discussions/10539). 😕

Yes I can ping github
/mnt/c/Users/h/Desktop/Coursera/UnixWorkbench/exercise/git/my-first-repo$ ping github.com
PING github.com (140.82.112.4) 56(84) bytes of data.
64 bytes from lb-140-82-112-4-iad.github.com (140.82.112.4): icmp_seq=1 ttl=44 time=59.7 ms
64 bytes from lb-140-82-112-4-iad.github.com (140.82.112.4): icmp_seq=2 ttl=44 time=61.2 ms
64 bytes from lb-140-82-112-4-iad.github.com (140.82.112.4): icmp_seq=3 ttl=44 time=61.4 ms
64 bytes from lb-140-82-112-4-iad.github.com (140.82.112.4): icmp_seq=4 ttl=44 time=66.6 ms

I have IPv6 connectivity. Maybe that's the reason?
 ifconfig
lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 1500
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0xfe<compat,link,site,host>
        loop  (Local Loopback)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wifi0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.0.1.5  netmask 255.255.255.0  broadcast 10.0.1.255
        inet6 2603:9000:b600:434a:e8da:ef07:b105:f9e5  prefixlen 64  scopeid 0x0<global>
        inet6 2603:9000:b600:434a:d23:30bd:2adf:d296  prefixlen 128  scopeid 0x0<global>
        inet6 fe80::e8da:ef07:b105:f9e5  prefixlen 64  scopeid 0xfd<compat,link,site,host>
        ether 00:91:9e:6c:11:a0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

**Reply by airtower-luna** [reply]
> I have IPv6 connectivity. Maybe that's the reason?

If it's IPv6 *only* that's a problem, and the `ifconfig` output looks like that.

The ping seemed to work fine with IPv4 though, was that from the same machine (container, VM, …)? :thinking:

**Reply by chilianlin87** [reply]
> > I have IPv6 connectivity. Maybe that's the reason?
> 
> If it's IPv6 _only_ that's a problem, and the `ifconfig` output looks like that.
> 
> The ping seemed to work fine with IPv4 though, was that from the same machine (container, VM, …)? 🤔

yes, from the same machine

**Reply by airtower-luna** [reply]
I guess you have a NAT-ed IPv4 connection over the 10.0.1.0/24 network? Then that part should be okay.

What's weird is that `ping` can resolve the hostname, but Git apparently can't. That doesn't really make sense. Either it was a transitory issue and Git works now, too, or Git somehow does name resolution differently from `ping` (which would be *very* weird, but I've seen plenty of weird). :thinking:

**Reply by chilianlin87** [reply]
> I guess you have a NAT-ed IPv4 connection over the 10.0.1.0/24 network? Then that part should be okay.
> 
> What's weird is that `ping` can resolve the hostname, but Git apparently can't. That doesn't really make sense. Either it was a transitory issue and Git works now, too, or Git somehow does name resolution differently from `ping` (which would be _very_ weird, but I've seen plenty of weird). 🤔

now when I run git push -u origin master
it give me error 
fatal: unable to access 'https://github.com/chilianlin87/my-first-repo.git/': Operation timed out after 300012 milliseconds with 0 out of 0 bytes received

**Reply by airtower-luna** [reply]
Even weirder. :sweat_smile: 

Timeout after a long wait without any data transmitted smells of router or firewall issues. But really it's impossible to tell for sure from remote. Depending on the kind of network you're on I'd check the relevant devices (router, modem, etc.) or contact people in charge (provider, network admin, …) of your uplink. If you can reach other sites just fine it's more likely to be an issue further away.

However, I'm wondering: How are you posting here? Are you using a different network?

**Reply by chilianlin87** [reply]
> sing a different netw

I am posting here in windows
I am trying to push in Ubuntu (bash actually)

**Reply by airtower-luna** [reply]
Dual boot on the same machine? VM? WSL? Something else? If they're on the same network that's critical information, because any difference could hint at the problem.

Ubuntu is a Linux distribution, Bash is a shell. Both important, but very different things. :wink:

**Reply by chilianlin87** [reply]
> Dual boot on the same machine? VM? WSL? Something else? If they're on the same network that's critical information, because any difference could hint at the problem.
> 
> Ubuntu is a Linux distribution, Bash is a shell. Both important, but very different things. 😉

WSL, on same machine, same network.
I am new to Linux, working on The Unix Workbench course on Coursera. I just type in Bash in start menu, entered bash there to practice.

**Reply by airtower-luna** [reply]
Hm, that makes it sound like a problem with the WSL network setup, but I don't use Windows myself so I'm not familiar with it. :sweat_smile: My first suspect would be the firewall though, see if it prevents to/from the WSL system (which is effectively a VM, AFAIK).

**Reply by chilianlin87** [reply]
> Hm, that makes it sound like a problem with the WSL network setup, but I don't use Windows myself so I'm not familiar with it. 😅 My first suspect would be the firewall though, see if it prevents to/from the WSL system (which is effectively a VM, AFAIK).

I installed git for windows as a temporary solution
At least I can push pull now, to continue the study.

**Reply by chilianlin87** [reply]
Thank you both so much for your help. I really appreciate.

**Reply by Diandrash** [reply]
Thank you its worked for me, this problem because i configure the ip v6

**Comment by chilianlin87** [comment]
This solution helped me. Here is the [link](https://askubuntu.com/questions/1192347/temporary-failure-in-name-resolution-on-wsl/1401317#1401317?newreg=fc95a6947d13420fb5f39420bf285388)
Inside WSL2, create or append file: /etc/wsl.conf

Put the following lines in the file in order to ensure the your DNS changes do not get blown away

sudo tee /etc/wsl.conf << EOF
[network]
generateResolvConf = false
EOF
In a cmd window (!!), run wsl --shutdown

Start WSL2

Run the following inside WSL2 (line with search is optional)

sudo rm -rf /etc/resolv.conf
sudo tee /etc/resolv.conf << EOF
search yourbase.domain.local
nameserver 8.8.8.8
nameserver 1.1.1.1
EOF

**Reply by tm-moumakoe** [reply]
thank you for this, i've been having the same problem

**Reply by huuga513** [reply]
Thanks, that really helps

**Reply by akhil-flycatch** [reply]
It worked, Thanks

**Comment by Hobart2967** [comment]
Yes, WSL Version 2 has some critical networking issue, I have solved it myself some time ago. Will have a look on how it was solved when I boot up my windows box next time and post it here. In Version 1 btw, everything works fine but is a bit slower. Maybe downgrading the distro may be of help for you as well
